### ISSUE TITLE
MaxPool1d Inconsistent Shape Checking

### ISSUE THREAD

**Issue by konstantinosKokos** [pytorch/pytorch#51954]
## 🐛 Bug

MaxPool1d raises RuntimeError due to shape (expected 3-dimensional tensor but got 2-dimensional tensor for argument #1 'self') when composed with a Linear Layer

## To Reproduce

Instantiate a MaxPool1d layer, and call with an arbitrary 2d tensor after passing it through a Linear layer.

```
>>> mp = MaxPool1d(2)
>>> x = torch.rand(32, 768)
>>> print(mp(x).shape)  # works
torch.Size([32, 384])
>>> f = Linear(768, 768)
>>> y = f(x)
>>> print(y.shape)  # works
torch.Size([32, 768])
>>> print(mp(y).shape)  # crashes
RuntimeError: Expected 3-dimensional tensor, but got 2-dimensional tensor for argument #1 'self' (while checking arguments for max_pool1d)
```

## Expected behavior

The layer should either also throw an error when called with x (arbitrary 2d tensor) or not throw an error at all when called with y (a tensor of shape identical to x)

cc @albanD @mruberry @jbschlosser @heitorschueroff @jianyuh @nikitaved @pearu @walterddr @IvanYashchuk

**Comment by mrshenli** [comment]
I confirm I can reproduce this error on master.  '1.8.0a0+4577f3f'

**Comment by mrshenli** [comment]
Hey @heitorschueroff, looks like #47065 is the last PR that touches `max_pool1d`. Do you know what could cause the reported error?

**Comment by mruberry** [comment]
The divergence is whether or not the tensor requires grad. In the above example the linear layer's output requires grad, the original input to maxpool1D does not.

This is because there's a divergence between the no indices and with indices max pool input checking. One allows only 3D inputs, the other allows 2D and 3D. I'm not sure how this worked historically. The documentation suggests the input must be 3D.

**Comment by heitorschueroff** [comment]
I vaguely remember coming across this problem when reviewing another PR. I think the fix was simple, I'll take a look.

**Comment by jbeck9** [comment]
I ran into this problem today. I'm going to leave this work around here for the time being. Not extremely familier with Pytorch, but I can't think of any reason a 1D max pooling layer should not be performed on a 2D tensor. The following class can be used to replace nn.MaxPool1D until the bug is fixed:

```
class MaxPool1d(nn.Module):
    def __init__(self, kernel, stride):
        super().__init__()
        self.max_pool_layer= nn.MaxPool1d(kernel,stride)
    
    def forward(self, x):
        return torch.squeeze(self.max_pool_layer(x.unsqueeze(0)))
```

**Comment by khushi-411** [comment]
Hello, there!
I think this issue is resolved in the latest master branch (`'1.13.0a0+git016fcca'`). Please see:
```python
In [20]: mp = torch.nn.MaxPool1d(2)

In [21]: x = torch.rand(32, 768)

In [22]: mp(x).shape
Out[22]: torch.Size([32, 384])

In [23]: f = torch.nn.Linear(768, 768)

In [24]: y = f(x)

In [25]: y.shape
Out[25]: torch.Size([32, 768])

In [26]: mp(y).shape
Out[26]: torch.Size([32, 384])
```
This issue can be closed as completed. Could you please re-check? Thanks!

**Comment by albanD** [comment]
Confirmed!
